src/components/Authenticator.js

```diff
@@ -1,30 +1,42 @@
-import Cookies from 'js-cookie';
+import {findDocumentQuery} from './database/Queries';
+import type {OAuthUserType, UserType} from './admin/FlowTypes';
 
-const mockAuth = {
-  isAdmin: function() {
-    const adminCookie = Cookies.get('admin');
-    return (adminCookie !== undefined);
-  },
-
-  isUser: function() {
-    const userCookie = Cookies.get('user');
-    return (userCookie !== undefined);
-  },
+const readUserDataFromDB = (user) => {
+  if (user === null) {
+    return Promise.resolve([]);
+  }
+  return findDocumentQuery('active-members', 'email', user.email);
+};
 
-  isInviter: function() {
-    const inviterCookie = Cookies.get('inviter');
-    return (inviterCookie !== undefined);
+const firebaseAuthenticator = {
+  isAdmin: function(user: OAuthUserType): Promise<boolean> {
+    let userDataPromise = readUserDataFromDB(user);
+    if (userDataPromise) {
+      return userDataPromise.then(
+          (userData: Array<UserType>) =>
+              Promise.resolve(userData.length > 0 && userData[0].isAdmin)
+          );
+    } else {
+      return Promise.resolve(false);
+    }
   },
 
-  logIn: function() {
-    Cookies.set('user', 'user');
+  isUser: function(user: OAuthUserType): Promise<boolean> {
+    let userDataPromise = readUserDataFromDB(user);
+    if (userDataPromise) {
+      return userDataPromise.then((userData: Array<UserType>) =>
+          userData.length > 0);
+    } else {
+      return Promise.resolve(false);
+    }
   },
 
-  logOut: function() {
-    Cookies.remove('user');
+  isInviter: function(user: OAuthUserType): boolean {
+    if (user === null) {
+      return false;
+    }
+    return user && user.email.split('@').pop() === 'google.com';
```

Comment:
this function doesn't return a Promise.

Reply:
Why should this function return Promise? It returns boolean as stated in function's flow signature.
We don't need to ask DB if currently logged in user is a Googler -- we can read it directly from user's email.
Could you please elaborate more on this comment?

Reply:
because all the other `is`-functions (`isAdmin` and `isUser`) return `Promise<boolean>`. It's a break in continuity.

Reply:
I agree with Leon that consistency in the interfaces of the functions has more value than an "optimizaton" where the returned type is more narrowly following the implementation.

Reply:
Done.